lib/games/matching.dart

```diff
@@ -0,0 +1,313 @@
+        child: _buildButton(s, index++),
+      ),
+    ]);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    int index = 0;
+    return LayoutBuilder(builder: (context, constraints) {
+      final hPadding = pow(constraints.maxWidth / 150.0, 2);
+      final vPadding = pow(constraints.maxHeight / 150.0, 2);
+
+      double maxWidth =
+          (constraints.maxWidth - hPadding * 2) / 2; //- middle_spacing;
+      double maxHeight = (constraints.maxHeight - vPadding * 2) / 8;
+
+      final buttonPadding = sqrt(min(maxWidth, maxHeight) / 5);
+
+      maxWidth -= buttonPadding * 2;
+      maxHeight -= buttonPadding * 2;
+      UnitButton.saveButtonSize(context, 1, maxWidth, maxHeight);
+      AppState state = AppStateContainer.of(context).state;
+      print('constraint Box:: $constraints');
+      return Container(
+        decoration: BoxDecoration(
+            color: Color(0xffA52A2A),
+            borderRadius: BorderRadius.only(
+                topRight: Radius.circular(30), topLeft: Radius.circular(30))),
+        child: CustomPaint(
+          foregroundPainter: DrawLine(
+              dottedOffset: _joinedLinedOffset,
+              startOffset: _startOffset,
+              endOffset: _updateOffset),
+          child: GestureDetector(
+              behavior: HitTestBehavior.opaque,
+              onScaleStart: (ScaleStartDetails str) {
+                Offset start = (context.findRenderObject() as RenderBox)
+                    .globalToLocal(str.focalPoint);
+                _onStart(start);
+              },
+              onScaleUpdate: (ScaleUpdateDetails det) {
+                Offset update = (context.findRenderObject() as RenderBox)
+                    .globalToLocal(det.focalPoint);
+                if (_isUnderCircle) _onUpdate(update);
+              },
+              onScaleEnd: _onEnd,
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceAround,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: <Widget>[
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: _question.map((s) {
+                      return _build(constraints, index++, s,
+                          AlignmentDirectional.centerEnd);
+                    }).toList(growable: false),
+                  ),
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: _answer.map((a) {
+                      return _build(constraints, index++, a,
+                          AlignmentDirectional.centerStart);
+                    }).toList(growable: false),
+                  ),
+                ],
+              )),
+        ),
+      );
+    });
+  }
+}
+
+class DottedCircle extends StatefulWidget {
+  final Function offsetOfCicle;
+  final BoxConstraints constraint;
+  final double height;
+  final GlobalKey keys;
```

Comment:
Why is this called keys if it is a single object. Should be key. Also pass it along to the parent in constructor - super(key: key)